Develop

MaritimeAst opened the pull request.

MaritimeAst committed:
Commit user firstTest

MaritimeAst committed:
Commit user tests

MaritimeAst committed:
Commit user tests

MaritimeAst committed:
Commit order tests

MaritimeAst committed:
Commit order tests

MaritimeAst committed:
add allure report

pomegranate282922 commented on src/main/java/models/User.java:
```diff
@@ -0,0 +1,39 @@
+public class User {
+
+
+    public String email;
```
Нарушение принципа инкапсуляции. Нужно использовать минимальный из возможных модификаторов доступа

pomegranate282922 commented on src/test/java/CreateUserValidationParametrizedTest.java:
```diff
@@ -0,0 +1,52 @@
+
+        assertEquals("Проверка корректности возвращаемого кода ошибки, при создании курьера без обязательных полей", statusCode, actualStatusCode);
+        assertEquals("Проверка корректности возвращаемого текста ошибки, при создании курьера без обязательных полей", message, actualMessage);
+    }
```
В негативных кейсах в случае пойманного бага тоже может создасться юзер. Его нужно удалить в этом случае.

pomegranate282922 reviewed (commented):
Все отлично - только бы ещё доделать удаление юзера в негативном кейсе его создания.

MaritimeAst committed:
fix after review
1) исправлен модификатор дрступа в UserClient
2) добавлено удаление пользователя в createUserValidationTest, в случае если в результате бага в негативном тесте пользователь будет добавлен
3) Добавлена задержка в 200мс для предотвращения появления ошибки 429

MaritimeAst commented:
Указанные недочеты исправлены.
fix after review
1) исправлен модификатор дрступа в UserClient
2) добавлено удаление пользователя в createUserValidationTest, в случае если в результате бага, в негативном тесте пользователь будет добавлен
3) Добавлена задержка в 200мс для предотвращения появления ошибки 429

pomegranate282922 reviewed (approved):
Зачёт!
